### PR TITLE
Add `renderToString` and `renderToStaticMarkup`

### PR DESCRIPTION
**Pull request by maxdeviant**
This PR adds support for the following `ReactDOMServer` methods:
- `renderToString`
- `renderToStaticMarkup`

I opted not to add the stream-based methods at this time (mostly cause I don't feel like typing the streams), but I can do that if it is desired.

### CODE REVIEW

File: src/React/Basic/DOM/Server.purs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+-- | Render a React element to its initial HTML. React will return an HTML string.
+-- | You can use this method to generate HTML on the server and send the markup
+-- | down on the initial request for faster page loads and to allow search engines
+-- | to crawl your pages for SEO purposes.
+-- |
+-- | If you call `ReactDOM.hydrate` on a node that already has this server-rendered
+-- | markup, React will preserve it and only attach event handlers, allowing you to
+-- | have a very performant first-load experience.
+foreign import renderToString :: JSX -> String
```
Comment (maxdeviant) [inline src/React/Basic/DOM/Server.purs:16]: Should these return an `Effect String` as opposed to just a `String`? I'm not sure if rendering to a string is side-effectful.

**Comment by maxdeviant** [comment]
I believe this PR resolves https://github.com/lumihq/purescript-react-basic/issues/63.

**Comment by i-am-the-slime** [comment]
2 hours ago? I wanted to upstream this just now:

```js
const reactDom = require("react-dom/server")
exports.renderToString = reactDom.renderToString
```

```purescript
foreign import renderToString ∷ JSX -> String
```

And I was wondering the same thing about effectfulness. I didn't model it that way out of laziness but I suppose it wouldn't really hurt either.

**Comment by maxdeviant** [comment]
You're right, it probably doesn't hurt to have an `Effect`, because these are most likely to be used at the top-level of a program.

But I'm more interested in whether or not it is correct to model them with or without an `Effect`, but perhaps there is no definitive consensus on that? :slightly_smiling_face:

File: src/React/Basic/DOM/Server.purs (hunk 1 above)
Comment (spicydonuts) [reply src/React/Basic/DOM/Server.purs:16]: Good question.. I don't think these functions ever preserve state (nothing gets mounted), so my guess is this is fine..

**Comment by spicydonuts** [comment]
Commented on it above as well.. I'm not really sure. It could throw, but that's not a normal path so maybe that doesn't matter.

**Comment by spicydonuts** [comment]
It'd be great to have the streaming functions as well, but that introduces Node dependencies. Which reminds me, `react-dom/server` is kind of a different package from `react-dom` as well.. but maybe neither of those things matter because it's a separate module and bundlers will ignore it.

**Comment by spicydonuts** [comment]
I've kind of been lazy about this whole thing because the server-side version of suspense is still up in the air and I was curious how that would change things. I'd guess it'd be through new functions though to preserve compatibility so it is more laziness than a real concern.

File: src/React/Basic/DOM/Server.purs (hunk 1 above)
Comment (maxdeviant) [reply src/React/Basic/DOM/Server.purs:16]: I'll leave the signature as-is then.

**Comment by maxdeviant** [comment]
@spicydonuts Are there any particular changes you'd like me to make to this PR?

FWIW, I think that introducing the streaming functions may fit better in a separate PR since, as you pointed out, it introduces Node dependencies.

**Comment by spicydonuts** [comment]
That's fine with me

**Comment by spicydonuts** [comment]
Thanks!

**Pull request merged**